tests/tests/swfs/avm2/filereference_save_and_browse/Test.as

```diff
@@ -0,0 +1,81 @@
+package {
+  import flash.display.Sprite;
+  public class Test extends Sprite { }
+}
+
+import flash.net.FileReference;
+import flash.net.FileFilter;
+import flash.events.Event;
+import flash.events.ProgressEvent;
+import flash.utils.setTimeout;
+
+var file = new FileReference();
+
+function dump(file) {
+    try {
+        trace("file.name: " + file.name);
+    } catch (e) {
+        trace("file.name threw: " + e);
+    }
+/* FIXME
+    try {
+        trace("file.size: " + file.size);
+    } catch (e) {
+        trace("file.size threw: " + e);
+    }
+*/
+    trace("file.data: " + file.data);
+    trace("");
+}
+
+function onopen(e) {
+    trace("open event");
+    dump(e.target);
+}
+
+function onprogress(e) {
+    trace("progress event");
+    trace(e.bytesLoaded + " / " + e.bytesTotal);
+    dump(e.target);
+}
+
+function oncomplete(e) {
+    trace("complete event");
+    dump(e.target);
+
+    file.removeEventListener(Event.SELECT, onselect);
+    file.removeEventListener(Event.COMPLETE, oncomplete);
+    file.addEventListener(Event.SELECT, onselect2);
+    file.addEventListener(Event.COMPLETE, oncomplete2);
+    file.browse([new FileFilter("debug-select-success", "*.txt")]);
```

Comment:
Might be simpler to just do something like the following instead of duplicating all the events.

```rust
if (saving) {
   file.browse()
}
saving = false;
```